### PR TITLE
Task 8 solution

### PR DESCRIPTION
**Pull request by maksumov**
# Task 8 Definition

Task defined at https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task8-docker-elastic-beanstalk/task.md

- [x] [Task 8 Prerequisites](https://github.com/maksumov/rs-cart-api/issues/2)

There are also two linked issues in the [another repository](https://github.com/maksumov/rss-shop-be) created:

- [x] [Task 8.1](https://github.com/maksumov/rss-shop-be/issues/8)
- [x] [Task 8.2](https://github.com/maksumov/rss-shop-be/issues/9)

# Solution URLs

1. App FE URL: https://d3sod2gr47frq2.cloudfront.net/
2. App PR: https://github.com/maksumov/nodejs-aws-fe/pull/10
3. maksumov-cart-api HTTP endpoint URL: http://maksumov-cart-api-dev.eu-west-1.elasticbeanstalk.com/
4. maksumov-cart-api HTTPS URL: https://u7dyaoudn4.execute-api.eu-west-1.amazonaws.com/dev/

# Self-check

- [x] Dockerfile is prepared, image is building. Image size is minimised to be [less than 500 MB](https://github.com/maksumov/rss-shop-be/issues/8#issuecomment-740025250).
image
- [x] [Dockerfile](https://github.com/maksumov/rs-cart-api/blob/task-8/Dockerfile) is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.
- [x] Folders are added to [.dockerignore](https://github.com/maksumov/rs-cart-api/blob/task-8/.dockerignore), with explanations. At least 2 big directories should be excluded from build context. [Elastic Beanstalk application is initialized](https://github.com/maksumov/rss-shop-be/issues/9#issuecomment-740209997).
image
- [x] [Environment is created](https://github.com/maksumov/rss-shop-be/issues/9#issuecomment-740213101) and the app is deployed to the AWS cloud. You must provide a link to your GitHub repo with Cart API service or PR with created Dockerfile and related configurations (**here it is**).
image
- [x] FE application is updated with Cart API endpoint. You must provide a [PR with updates in your FE repository](https://github.com/maksumov/nodejs-aws-fe/pull/10) and OPTIONALLY [link to deployed front-end app](https://d3sod2gr47frq2.cloudfront.net/) which makes proper API calls to your [Cart service](https://u7dyaoudn4.execute-api.eu-west-1.amazonaws.com/dev/ping).

All tasks completed. Expected score: 5/5.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Remove Git history - it doesn't needed in container
```
Comment (kirill-ianchuk) [inline .dockerignore:1]: What about `dist` folder?

Comment (maksumov) [reply .dockerignore:1]: I don't understand the question. Do you suggest adding the dist folder here?

Comment (maksumov) [reply .dockerignore:1]: Thanks for the suggestion. I implemented it

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+FROM mhart/alpine-node:12 AS builder
+
+WORKDIR /usr/src/app
+
+# Install development dependencies
+COPY package*.json ./
+RUN npm ci
+
+# Generate dist
+COPY . .
```
Comment (maksumov) [inline Dockerfile]: It should be worth to change this line on `COPY ./src ./src`

**Pull request merged**